scripts/simulation.py

```diff
@@ -0,0 +1,96 @@
+"""
+The purpose of this script is to provide an interface for the user to use 
+hera_sim to create, populate, and save to disk a UVData object.
+"""
+import hera_sim
+import sys
+import os
+import yaml
+from bda import bda_tools
+
+# syntax for running from command line:
+# python simulation.py path_to_config path_to_save verbose 
+# anything else we need to specify?
+# notes: 
+# path_to_config should be an absolute path to a configuration YAML
+# the config YAML will completely specify the instrument and sim component
+# parameters (to be applied with defaults and run_sim)
+# the decision to perform BDA or not (and which parameters to use) will also
+# be made in the config file
+#
+# path_to_save should be an absolute path specifying the save location with an extension
+#
+# how to handle verbose?
+
+# get configuration path, simulation parameter path, save path
+config = sys.argv[1]
+
+# confirm that config and simulation parameter paths exist
+assert os.path.exists(config), \
+        "The config file could not be found. Please ensure a path to the file " \
+        "exists. The provided path is: {}".format(config)
+
+# load in config
+with open(config, 'r') as f:
+    yaml_contents = yaml.load(f.read(), Loader=yaml.FullLoader)
+
+# verbose option?
+verbose = True if "verbose" in sys.argv else False
+
+# extract parameters for saving to disk
+filing_params = yaml_contents["filing"]
+outfile = os.path.join(filing_params["outdir"], filing_params["outfile_name"])
+save_path = "{}.{}".format(outfile, filing_params["output_format"])
+clobber = filing_params["clobber"]
+
+# XXX add a check to make sure the save type is OK?
+
+# determine whether to use season defaults
+defaults = yaml_contents.get("defaults", {})
+apply_defaults = defaults.pop("apply_defaults", False)
+if apply_defaults and defaults:
+    if isinstance(defaults["default_config"], str):
+        hera_sim.defaults.set(defaults["default_config"])
+    elif isinstance(defaults["default_config"], dict):
+        hera_sim.defaults.set(**defaults["default_config"])
+    else:
+        raise ValueError("If you wish to override function defaults, then " \
+                "you must do so by either specifying a path to the default " \
+                "configuration file or specifying an appropriately formatted " \
+                "dictionary of default values.")
+
+
+# TODO: figure out how to specify antennas
+# extract instrument parameters
+antennas = sim_params.pop("antennas")
+instrument_params = {"antennas" : antennas}
+for parameter in ("freq", "time", ):
+    for key, value in yaml_contents[parameter].items():
+        instrument_params[key] = value
+
+sim = hera_sim.Simulator(**instrument_params)
+
+# extract simulation parameters
+sim_params = {}
+for component in ("systematics", "analysis", ):
```

Comment:
The tuple should be reordered so that systematics are simulated last (or we can revert back to having `run_sim` apply effects in a fixed order).

Reply:
what are some example entries in "systematics" and "analysis"?

Reply:
This is what has been the bottleneck in finishing development for this branch. I'm not exactly sure what should go in each one, and how it should be handled, so I figured we should have a discussion about this. Additionally, as I was preparing this, I noticed that there is a lot of redundancy between the ways that `defaults` and `run_sim` are handled; I suppose this is OK if we're fine with letting `run_sim` override some of the settings in `defaults`, but I can see that as a potential cause for confusion with end-users.

I'm thinking that "systematics" should contain information about gains and crosstalk at the very least, as well as maybe information about rfi and thermal noise. "analysis" should definitely contain information about the EoR signal and foregrounds. I was thinking it could be a good idea to separate things in a way that "systematics" and "analysis" cover all the bases for simulation specifics, while "defaults" handles things more related to season-specific quantities (like `omega_p`). It's a little tricky, though, since there are some quantities that could easily fall into multiple categories (for example, the bandpass interpolator could live in "systematics" just as well as "defaults", and the sky temperature model live in "analysis" or "defaults").

So the question of where to put things ("defaults", "analysis", or "systematics") is still open. I'd appreciate any suggestions you have on the topic.

Reply:
Hmmm. Yeah good point, that there's a lot of crossover between defaults and the other categories. I'm not sure of the best way to handle this. One way could be to assume in this case that `defaults` literally has to be a string path or identifier, and that these parameters really just overwrite those defaults at some places, and add some filing information. 

If most of the parameters are the same between defaults and this config, you can use the same function to interpret them (or at least most of them).